src/vm/fastserializableobject.h

```diff
@@ -25,6 +25,14 @@ class FastSerializableObject
 
     // Get the type name for the current object.
     virtual const char* GetTypeName() = 0;
+
+    int GetObjectVersion() const { return m_objectVersion; }
+    int GetMinReaderVersion() const { return m_minReaderVersion; }
+
+protected:
+
+    int m_objectVersion = 1;
```

Comment:
Rather than having protected member fields, can you make these private and put them behind protected Set methods?